unit/testing-utils/generic_utils.cpp

```diff
@@ -0,0 +1,39 @@
+/*******************************************************************\
+
+ Module: Unit test utilities
+
+ Author: DiffBlue Limited. All rights reserved.
+
+\*******************************************************************/
+
+#include <util/ui_message.h>
+#include <java_bytecode/generate_java_generic_type.h>
+#include "generic_utils.h"
+#include "catch.hpp"
+#include "require_type.h"
+
+/// Generic a specalised version of a Java generic class and add it to the
+/// symbol table.
+/// \param example_type: A reference type that is a specalised generic to use
+/// as the base for the specalised version (e.g. a variable of type
+/// `Generic<Integer>`
+/// \param new_symbol_table: The symbol table to store the new type in
+void generic_utils::specalise_generic(
+  const java_generic_typet &example_type,
+  symbol_tablet &new_symbol_table)
+{
+  // Should be a fully instantiated generic type
+  REQUIRE(
+    std::all_of(
+      example_type.generic_type_variables().begin(),
+      example_type.generic_type_variables().end(),
+      [](const java_generic_parametert &generic_parameter) {
+        return is_java_generic_inst_parameter(generic_parameter);
```

Comment:
No need for lambda as `is_java_generic_inst_parameter` is already of the type you want?

Reply:
I think it is needed: note the **`inst`** (meaning an instantiated parameter like `Integer` rather than just  a generic parameter like `T`

Reply:
Me being a dope - now addressed